src/main/java/Penguin.java

```diff
@@ -0,0 +1,238 @@
+                }
+
+                Task t = tasks.get(id - 1);
+                System.out.println(line);
+                System.out.println("Nice! I've marked this task as done:");
+                System.out.println(t.markAsDone());
+                System.out.println(line);
+
+                try {
+                    save.save(tasks);
+                } catch (IOException e) {
+                    // should not happen, unless user deletes file mid run
+                    return false;
+                }
+
+                return false;
+            }
+            case UNMARK -> {
+                int id = parseInt(userInput, 7);
+                if (!isValidIndex(line, tasks, id)) {
+                    return false;
+                }
+
+                Task t = tasks.get(id - 1);
+                System.out.println(line);
+                System.out.println("OK, I've marked this task as not done yet:");
+                System.out.println(t.markAsNotDone());
+                System.out.println(line);
+
+                try {
+                    save.save(tasks);
+                } catch (IOException e) {
+                    // should not happen, unless file deleted mid run
+                    return false;
+                }
+
+                return false;
+            }
+            case DELETE -> {
+                String rest = userInput.substring(7);
+                int id;
+                try {
+                    id = Integer.parseInt(rest);
+                } catch (NumberFormatException e) {
+                    throw new PenguinException("Format: delete <task id>");
+                }
+                if (id < 1 || id > tasks.size()) {
+                    throw new PenguinException("Task id out of range! (1..." + tasks.size() + ")");
+                }
+                Task removed = tasks.remove(id - 1);
+                System.out.println(line);
+                System.out.println("Noted. I've removed this task:");
+                System.out.println("  " + removed.toString());
+                System.out.println("Now you have " + tasks.size() + " task(s) in the list!");
+                System.out.println(line);
+
+                try {
+                    save.save(tasks);
+                } catch (IOException e) {
+                    // should not happen, unless file deleted mid run
+                    return false;
+                }
+
+                return false;
+            }
+            case DEADLINE -> {
+                String body = userInput.substring(9);
+                String[] parts = body.split("\\s*/by\\s*", 2);
+                // check if valid
+                if (parts.length != 2 || parts[0].isBlank() || parts[1].isBlank()) {
+                    throw new PenguinException("Format: deadline <description> /by <when>");
+                }
+                Task t = new Deadline(parts[0], parts[1]);
+                tasks.add(t);
+                printAdded(line, t, tasks.size());
```

Comment:
Can consider reaming this variable to present tense
```suggestion
                printAdd(line, t, tasks.size());
```